.gitignore

```diff
@@ -1,3 +1,4 @@
 /node_modules/
 
-/parser.js
\ No newline at end of file
+/parser.js
+/test/UserAgentBag-types.test.js
```

Comment:
Instead of actually compiling this, could we use `tsc --noEmit` to check the result without actually compiling it?

Reply:
Well - the thing with type definitions in `index.d.ts` is that TypeScript isn't able to check that they actually match the JS code. The definitions could be misspelt or whatever, and `tsc` will just happily use it when transpiling any dependencies.

With this diff, the `pretest` step runs `tsc` to produce this file (assuming `index.d.ts` actually matches the JS), then the `test` step executes this file. Only by executing this file can we test that users will actually be able to import the module via TypeScript.